system/java_certs.py

```diff
@@ -0,0 +1,199 @@
+    (rc, out, err) = self.module.run_command(test_cmd)
+    if rc == 0:
+        return True
+    return False
+
+def import_cert_url(module, url, port, keystore_path, keystore_pass, alias):
+    fetch_cmd = "keytool -printcert -rfc -sslserver %s:%s" % (url, port)
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Fetch SSL certificate from remote host.
+    (rc, fetch_out, fetch_err) = module.run_command(fetch_cmd, check_rc=True)
+
+    # Use remote certificate from remote host and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, data=fetch_out, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def import_cert_path(module, path, keystore_path, keystore_pass):
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -file '%s'" % (keystore_path, keystore_pass, path)
+
+    # Use local certificate from local path and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    del_cmd = "keytool -delete -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Delete SSL certificate from keystore
+    (rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    return module.exit_json(changed=True, msg=del_out,
+        rc=rc, cmd=del_cmd, stdout_lines=del_out)
+
+def test_keytool(module):
+    ''' Test if keytool is actuall executable or not '''
+    test_cmd = "keytool"
+
+    (rc, del_out, del_err) = module.run_command(test_cmd, check_rc=True)
+
+def test_keystore(module, keystore_path):
+    ''' Check if we can access keystore as file or not '''
+    if not os.path.exists(keystore_path) and not os.path.isfile(keystore_path):
+        ## Keystore doesn't exist we want to create it
+        return module.fail_json(msg="Module require existing keystore at keystore_path '%s'" % keystore_path)
+
+def main():
+    argument_spec = dict()
+    argument_spec.update(dict(
```

Comment:
Don't see the point of updating an empty dict rather than just assigning the populated dict to argument_spec directly